File: leetcode/algorithms/747_largest_number_at_least_twice_of_others/main.py
from typing import List


class LargestNumberAtLeastTwiceOfOthers:
    def dominantIndex(self, nums: List[int]) -> int:
        max: int = -1
        second: int = -1
        index: int = -1

        for i in range(0, len(nums)):
            if nums[i] > max:
                second = max
                max = nums[i]
                index = i
            elif nums[i] > second:
                second = nums[i]

        return index if max >= second * 2 else -1


    # Solution
    def solution(self, nums: List[int]) -> int:
        max: int = -1
        index: int = -1
        second: int = -1

        for i in range(0, len(nums)):
            if nums[i] > max:
                second = max
                max = nums[i]
                index = i
            elif nums[i] > second:
                second = nums[i]

        return index if second * 2 <= max else -1

File: leetcode/algorithms/747_largest_number_at_least_twice_of_others/main.dart
class LargestNumberAtLeastTwiceOfOthers {
  int dominantIndex(List<int> nums) {
    int max = -1;
    int second = -1;
    int index = -1;

    for (int i = 0; i < nums.length; i++) {
      if (nums[i] > max) {
        second = max;
        max = nums[i];
        index = i;
      } else if (nums[i] > second) {
        second = nums[i];
      }
    }

    return max >= second * 2 ? index : -1;
  }


  // Solution
  int solution(List<int> nums) {
    int max = -1, index = -1, second = -1;

    for (int i = 0; i < nums.length; i++) {
      if (nums[i] > max) {
        second = max;
        max = nums[i];
        index = i;
      } else if (nums[i] > second) {
        second = nums[i];
      }
    }

    return second * 2 <= max ? index : -1;
  }
}
